### PR TITLE
Updated repository structure and added strict types.

### LINKED ISSUE #3: Update repo configs from https://getscaffold.dev/
Download the latest code from [Scaffold](https://github.com/AlexSkrypnyk/scaffold) repo (actual code, not the release).

- Update `composer.json` with the structure from Scaffold. Preserve the values of the existing fields, but order the fields as per Scaffold's `composer.json`. Add missing fields.
- Add Renovate config from Scaffold
- Add/update all GitHub actions and configs to be per Scaffold. **Note that the `test` GHA workflow should be used and CircleCI should be removed.**
- Update readme to use the structure from Scaffold, including the generated logo (circle). Transfer the usage examples into the readme structure from Scaffold
- Add PHPStan config and fix all newly found violations
- Reformat `composer.json` to have indents and sorting as per the Scaffold.

Make sure that CI passes.

Similar task for reference: https://github.com/AlexSkrypnyk/shell-variables-extractor/issues/17

### PR DESCRIPTION
**Pull request by tannguyen04**
closes #3

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  release-drafter:
+    permissions:
+      contents: write
+      pull-requests: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: release-drafter/release-drafter@v5
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  release-php:
```
Comment (AlexSkrypnyk) [inline .github/workflows/release.yml]: this `release-php` is used to release binaries - this is for the case when the project is a PHP command. this job would package it into a binary.

for this project,we do not need this as this project is just a class.

please remove

File: composer.json [hunk 2]
```diff
@@ -36,5 +33,20 @@
...
+    "scripts": {
+        "lint": [
+            "phpcs",
+            "phpmd --exclude vendor,vendor-bin,node_modules . text phpmd.xml",
+            "phpstan"
+        ],
+        "lint:fix": "phpcbf",
+        "test": "phpunit --no-coverage",
+        "test:coverage": "XDEBUG_MODE=coverage phpunit",
+        "build": [
```
Comment (AlexSkrypnyk) [inline composer.json]: please remove this `build` step - it is not used here

**Review by AlexSkrypnyk — CHANGES_REQUESTED** [review]
@tannguyen04 
could you please address my comments. thank you

**Comment by tannguyen04** [comment]
Hi @AlexSkrypnyk 
I updated code as your comment.

**Pull request merged**